From c094959b558b100832de439a7afe7d625aa228b1 Mon Sep 17 00:00:00 2001
From: denglei <denglei@eswincomputing.com>
Date: Tue, 2 Sep 2025 10:35:13 +0800
Subject: [PATCH 499/540] fix:Optimize memory usage in npu drivers.

Changelogs:
	1.reduce NPU driver requests for continuous memory.
	2.modify the npu code format.

Signed-off-by: denglei <denglei@eswincomputing.com>
---
 drivers/soc/eswin/ai_driver/dsp/dsp_ioctl.c  | 225 ++++-------
 drivers/soc/eswin/ai_driver/dsp/dsp_main.h   |   2 -
 drivers/soc/eswin/ai_driver/dsp/dsp_pool.c   |   6 +-
 drivers/soc/eswin/ai_driver/npu/dla_buffer.c |   7 +-
 drivers/soc/eswin/ai_driver/npu/dsp.c        | 374 +++++++------------
 drivers/soc/eswin/ai_driver/npu/engine.c     |  89 ++---
 drivers/soc/eswin/ai_driver/npu/parse_dep.c  |  75 ++--
 7 files changed, 255 insertions(+), 523 deletions(-)

diff --git a/drivers/soc/eswin/ai_driver/dsp/dsp_ioctl.c b/drivers/soc/eswin/ai_driver/dsp/dsp_ioctl.c
index b7d76b3a2cf5..147109322527 100644
--- a/drivers/soc/eswin/ai_driver/dsp/dsp_ioctl.c
+++ b/drivers/soc/eswin/ai_driver/dsp/dsp_ioctl.c
@@ -106,8 +106,7 @@ static long dsp_ioctl_load_op(struct file *flip, dsp_ioctl_load_s __user *arg)
 		goto err;
 	}
 
-	if (copy_from_user(op_name, (void *)dsp_load.op_name,
-			   OPERATOR_NAME_MAXLEN)) {
+	if (copy_from_user(op_name, (void *)dsp_load.op_name, OPERATOR_NAME_MAXLEN)) {
 		ret = -EFAULT;
 		goto err;
 	}
@@ -118,8 +117,7 @@ static long dsp_ioctl_load_op(struct file *flip, dsp_ioctl_load_s __user *arg)
 			ret = -ENOMEM;
 			goto err;
 		}
-		if (copy_from_user(op_dir, (void *)dsp_load.op_lib_dir,
-				   OPERATOR_DIR_MAXLEN)) {
+		if (copy_from_user(op_dir, (void *)dsp_load.op_lib_dir, OPERATOR_DIR_MAXLEN)) {
 			ret = -EFAULT;
 			goto err;
 		}
@@ -171,8 +169,7 @@ static u32 dsp_get_dma_addr(struct es_dsp *dsp, int fd,
 		return 0;
 	}
 
-	dma_addr = dev_mem_attach(dmabuf, dsp->dev, DMA_BIDIRECTIONAL,
-				  &map_buf->attach);
+	dma_addr = dev_mem_attach(dmabuf, dsp->dev, DMA_BIDIRECTIONAL, &map_buf->attach);
 	if (!dma_addr) {
 		dsp_err("dev mem attach fd=%d failed.\n", fd);
 		map_buf->attach = NULL;
@@ -223,11 +220,9 @@ int dsp_unmap_dmabuf(struct dsp_file *dsp_file, struct dsp_dma_buf **buf,
 			kfree(buf[i]);
 			buf[i] = NULL;
 		} else {
-			entry = container_of(buf[i], struct dsp_dma_buf_ex,
-					     buf);
-			dsp_debug("%s, %d, fd=%d, handl cnt=%d.\n", __func__,
-				  __LINE__, buf[i]->fd,
-				  kref_read(&entry->handle.refcount));
+			entry = container_of(buf[i], struct dsp_dma_buf_ex, buf);
+			dsp_debug("%s, %d, fd=%d, handl cnt=%d.\n", __func__, __LINE__, buf[i]->fd,
+					  kref_read(&entry->handle.refcount));
 			kernel_handle_decref(&entry->handle);
 		}
 	}
@@ -246,13 +241,10 @@ struct dsp_dma_buf *dsp_get_dma_buf_ex(struct dsp_file *dsp_file, int memfd)
 	mutex_lock(&dsp_file->xrray_lock);
 	entry = xa_load(&dsp_file->buf_xrray, memfd);
 	if (entry) {
-		dsp_debug("%s, %d. entry->fd=%d.\n", __func__, __LINE__,
-			  entry->buf.fd);
-		handle = find_kernel_handle(&dsp_file->h, entry->handle.fd,
-					    DSP_DMABUF_HANDLE_MAGIC);
-		dsp_debug("%s, %d, handle_fd=0x%x, refcount=%d.\n", __func__,
-			  __LINE__, entry->handle.fd,
-			  kref_read(&handle->refcount));
+		dsp_debug("%s, %d. entry->fd=%d.\n", __func__, __LINE__, entry->buf.fd);
+		handle = find_kernel_handle(&dsp_file->h, entry->handle.fd, DSP_DMABUF_HANDLE_MAGIC);
+		dsp_debug("%s, %d, handle_fd=0x%x, refcount=%d.\n", __func__, __LINE__, entry->handle.fd,
+				  kref_read(&handle->refcount));
 	}
 	mutex_unlock(&dsp_file->xrray_lock);
 
@@ -260,16 +252,14 @@ struct dsp_dma_buf *dsp_get_dma_buf_ex(struct dsp_file *dsp_file, int memfd)
 		dsp_debug("%s, %d, slow path.\n", __func__, __LINE__);
 		dma_entry = kzalloc(sizeof(struct dsp_dma_buf), GFP_KERNEL);
 		if (!dma_entry) {
-			dsp_err("%s, %d. alloc dsp device buf struct err.\n",
-				__func__, __LINE__);
+			dsp_err("%s, %d. alloc dsp device buf struct err.\n", __func__, __LINE__);
 			goto err;
 		}
 		addr = dsp_get_dma_addr(dsp_file->dsp, memfd, dma_entry);
 		dma_entry->fd = -1;
 	} else {
-		dsp_debug("%s, %d, fast path, dma fd=%d, iova=0x%x.\n",
-			  __func__, __LINE__, entry->buf.fd,
-			  entry->buf.dma_addr);
+		dsp_debug("%s, %d, fast path, dma fd=%d, iova=0x%x.\n", __func__, __LINE__, entry->buf.fd,
+				  entry->buf.dma_addr);
 		dma_entry = &entry->buf;
 		addr = dma_entry->dma_addr;
 	}
@@ -300,16 +290,14 @@ static int dsp_ioctl_set_flat(struct dsp_file *dsp_file, dsp_ioctl_task_s *req,
 		dsp_debug("%s, i=%d, new fd=%d, offset=0x%x.\n", __func__, i,
 			  (int)req->task.dspBuffers[i].memFd,
 			  req->task.dspBuffers[i].offset);
-		entry = dsp_get_dma_buf_ex(dsp_file,
-					   req->task.dspBuffers[i].memFd);
+		entry = dsp_get_dma_buf_ex(dsp_file, req->task.dspBuffers[i].memFd);
 		if (entry == NULL) {
 			dsp_err("getdmabuf failed curr idx:%d, cfgcnt:%d, incnt:%d\n",i,
 			req->task.bufferCntCfg, req->task.bufferCntInput);
 			goto err;
 		}
 		dma_entry[i] = entry;
-		flat->buffers[i].addr =
-			entry->dma_addr + req->task.dspBuffers[i].offset;
+		flat->buffers[i].addr = entry->dma_addr + req->task.dspBuffers[i].offset;
 		flat->buffers[i].size = req->task.dspBuffers[i].size;
 		dsp_debug("%s, i=%d, addr=0x%x, len=0x%x\n", __func__, i,
 			  flat->buffers[i].addr, flat->buffers[i].size);
@@ -325,8 +313,7 @@ static struct dsp_user *dsp_find_user_by_fd(struct dsp_file *dsp_file, int fd)
 	struct khandle *user_handle;
 	struct dsp_user *user;
 
-	user_handle =
-		find_kernel_handle(&dsp_file->h, fd, DSP_USER_HANDLE_MAGIC);
+	user_handle = find_kernel_handle(&dsp_file->h, fd, DSP_USER_HANDLE_MAGIC);
 	if (!user_handle) {
 		dsp_err("cannot find dsp operator for %d.\n", fd);
 		return NULL;
@@ -347,11 +334,9 @@ static int dsp_set_task_req(struct es_dsp *dsp, dsp_request_t *dsp_req,
 
 	buffer_count = task->task.bufferCntCfg + task->task.bufferCntInput +
 		       task->task.bufferCntOutput;
-	dma_len = sizeof(struct es_dsp_flat1_desc) +
-		  buffer_count * sizeof(es_dsp_buffer);
+	dma_len = sizeof(struct es_dsp_flat1_desc) + buffer_count * sizeof(es_dsp_buffer);
 
-	flat = (struct es_dsp_flat1_desc *)dsp_alloc_flat_mem(dsp, dma_len,
-							      &dma_addr);
+	flat = (struct es_dsp_flat1_desc *)dsp_alloc_flat_mem(dsp, dma_len, &dma_addr);
 	if (!flat) {
 		dsp_err("cannot dma alloc mem for desc.\n");
 		ret = -ENOMEM;
@@ -365,8 +350,7 @@ static int dsp_set_task_req(struct es_dsp *dsp, dsp_request_t *dsp_req,
 	dsp_req->flat_virt = (void *)flat;
 	dsp_req->prio = task->task.priority;
 	if (dsp_req->prio >= DSP_MAX_PRIO) {
-		dsp_err("%s, %d, dsp request prio = %d is err.\n", __func__,
-			__LINE__, dsp_req->prio);
+		dsp_err("%s, %d, dsp request prio = %d is err.\n", __func__, __LINE__, dsp_req->prio);
 		dsp_free_flat_mem(dsp, dma_len, (void *)flat, dma_addr);
 		return -EINVAL;
 	}
@@ -375,8 +359,7 @@ static int dsp_set_task_req(struct es_dsp *dsp, dsp_request_t *dsp_req,
 	flat->num_buffer = buffer_count;
 
 	flat->input_index = task->task.bufferCntCfg;
-	flat->output_index =
-		task->task.bufferCntCfg + task->task.bufferCntInput;
+	flat->output_index = task->task.bufferCntCfg + task->task.bufferCntInput;
 	return 0;
 }
 
@@ -400,10 +383,8 @@ static void dsp_user_async_req_complete(struct device *dev, dsp_request_t *req)
 	}
 	user = user_req->user;
 	dsp_file = user_req->dsp_file;
-	dsp_unmap_dmabuf(dsp_file, user_req->dma_entry,
-			 user_req->dma_buf_count);
-	dsp_free_flat_mem(dsp, req->flat_size, req->flat_virt,
-			  req->dsp_flat1_iova);
+	dsp_unmap_dmabuf(dsp_file, user_req->dma_entry, user_req->dma_buf_count);
+	dsp_free_flat_mem(dsp, req->flat_size, req->flat_virt, req->dsp_flat1_iova);
 	dsp_debug("%s, %d.\n", __func__, __LINE__);
 	kernel_handle_decref(&user_req->handle);
 	kernel_handle_decref(&user->h);
@@ -413,8 +394,7 @@ static void dsp_user_async_req_complete(struct device *dev, dsp_request_t *req)
 
 static void dsp_async_task_release(struct khandle *handle)
 {
-	struct dsp_user_req_async *user_req =
-		container_of(handle, struct dsp_user_req_async, handle);
+	struct dsp_user_req_async *user_req = container_of(handle, struct dsp_user_req_async, handle);
 	struct dsp_user *user;
 	struct es_dsp *dsp;
 
@@ -444,8 +424,7 @@ static void dsp_hw_complete_task(struct device *dev, dsp_request_t *req)
 
 	spin_lock_irqsave(&dsp_file->async_ll_lock, flags);
 	if (dsp_file->h.fd != INVALID_HANDLE_VALUE && async_task->need_notify) {
-		list_add_tail(&async_task->async_ll,
-			      &dsp_file->async_ll_complete);
+		list_add_tail(&async_task->async_ll, &dsp_file->async_ll_complete);
 		spin_unlock_irqrestore(&dsp_file->async_ll_lock, flags);
 
 		wake_up_interruptible(&dsp_file->async_ll_wq);
@@ -486,8 +465,7 @@ static struct dsp_user_req_async *dsp_set_task_info(struct dsp_file *dsp_file,
 		       task->task.bufferCntOutput;
 
 	user_req = kzalloc(sizeof(struct dsp_user_req_async) +
-				   sizeof(struct dsp_dma_buf *) * buffer_count,
-			   GFP_KERNEL);
+					   sizeof(struct dsp_dma_buf *) * buffer_count, GFP_KERNEL);
 	if (!user_req) {
 		kernel_handle_decref(&user->h);
 		dsp_err("kmalloc dsp request struct error.\n");
@@ -543,8 +521,7 @@ static struct dsp_user_req_async *dsp_set_task_info(struct dsp_file *dsp_file,
 	return NULL;
 }
 
-static void dsp_free_task(struct dsp_file *dsp_file,
-			  struct dsp_user_req_async *user_req)
+static void dsp_free_task(struct dsp_file *dsp_file, struct dsp_user_req_async *user_req)
 {
 	struct dsp_dma_buf **dma_entry = user_req->dma_entry;
 	struct es_dsp *dsp = dsp_file->dsp;
@@ -555,16 +532,14 @@ static void dsp_free_task(struct dsp_file *dsp_file,
 	if (dma_entry) {
 		dsp_unmap_dmabuf(dsp_file, dma_entry, buffer_count);
 	}
-	dsp_free_flat_mem(dsp, dsp_req->flat_size, dsp_req->flat_virt,
-			  dsp_req->dsp_flat1_iova);
+	dsp_free_flat_mem(dsp, dsp_req->flat_size, dsp_req->flat_virt, dsp_req->dsp_flat1_iova);
 
 	kernel_handle_release_family(&user_req->handle);
 	kernel_handle_decref(&user_req->handle);
 	kernel_handle_decref(&user->h);
 }
 
-static long dsp_ioctl_submit_tsk_async(struct file *flip,
-				       dsp_ioctl_task_s __user *arg)
+static long dsp_ioctl_submit_tsk_async(struct file *flip, dsp_ioctl_task_s __user *arg)
 {
 	struct dsp_file *dsp_file = flip->private_data;
 	struct es_dsp *dsp = dsp_file->dsp;
@@ -583,16 +558,6 @@ static long dsp_ioctl_submit_tsk_async(struct file *flip,
 
 	// using reserved for op_idx
 	dsp->op_idx = task->task.reserved;
-	if ((dsp_perf_enable || dsp->perf_enable) &&
-	    dsp->op_idx < MAX_DSP_TASKS) {
-		dsp->op_perf[dsp->op_idx].OpStartCycle =
-			0;  //get_perf_timer_cnt();
-		dsp->op_perf[dsp->op_idx].Die = dsp->numa_id;
-		dsp->op_perf[dsp->op_idx].CoreId = dsp->process_id;
-		dsp->op_perf[dsp->op_idx].OpIndex = dsp->op_idx;
-		dsp->op_perf[dsp->op_idx].OpType =
-			dsp->process_id + 7;  // IDX_DSP0
-	}
 
 	if (!try_module_get(THIS_MODULE)) {
 		dsp_err("%s, %d, cannot get module.\n", __func__, __LINE__);
@@ -625,9 +590,7 @@ static long dsp_ioctl_submit_tsk_async(struct file *flip,
 	return ret;
 }
 
-static long
-dsp_ioctl_process_complete_task(struct file *flip,
-				dsp_ioctl_async_process_s __user *arg)
+static long dsp_ioctl_process_complete_task(struct file *flip, dsp_ioctl_async_process_s __user *arg)
 {
 	struct dsp_file *dsp_file = flip->private_data;
 	struct es_dsp *dsp = dsp_file->dsp;
@@ -638,8 +601,7 @@ dsp_ioctl_process_complete_task(struct file *flip,
 	struct dsp_user_req_async *async_task = NULL;
 	unsigned long flags;
 
-	if (copy_from_user(&query_task, arg,
-			   sizeof(dsp_ioctl_async_process_s))) {
+	if (copy_from_user(&query_task, arg, sizeof(dsp_ioctl_async_process_s))) {
 		ret = -EINVAL;
 		return ret;
 	}
@@ -648,8 +610,7 @@ dsp_ioctl_process_complete_task(struct file *flip,
 		return -EINVAL;
 	}
 	query_task.return_num = 0;
-	task = kzalloc(query_task.task_num * sizeof(dsp_task_status_s),
-		       GFP_KERNEL);
+	task = kzalloc(query_task.task_num * sizeof(dsp_task_status_s), GFP_KERNEL);
 	if (!task) {
 		dsp_err("alloc memory for dsp_task_status error.\n");
 		return -ENOMEM;
@@ -659,14 +620,12 @@ dsp_ioctl_process_complete_task(struct file *flip,
 	if (list_empty_careful(&dsp_file->async_ll_complete)) {
 		spin_unlock_irqrestore(&dsp_file->async_ll_lock, flags);
 		if (query_task.timeout < 0) {
-			ret = wait_event_interruptible(
-				dsp_file->async_ll_wq,
-				!list_empty(&dsp_file->async_ll_complete));
+			ret = wait_event_interruptible(dsp_file->async_ll_wq,
+								!list_empty(&dsp_file->async_ll_complete));
 		} else {
-			ret = wait_event_interruptible_timeout(
-				dsp_file->async_ll_wq,
-				!list_empty(&dsp_file->async_ll_complete),
-				msecs_to_jiffies(query_task.timeout));
+			ret = wait_event_interruptible_timeout(dsp_file->async_ll_wq,
+								!list_empty(&dsp_file->async_ll_complete),
+								msecs_to_jiffies(query_task.timeout));
 		}
 	} else {
 		spin_unlock_irqrestore(&dsp_file->async_ll_lock, flags);
@@ -701,8 +660,7 @@ dsp_ioctl_process_complete_task(struct file *flip,
 	ret = 0;
 	if (query_task.return_num != 0) {
 		if (copy_to_user((void __user *)(query_task.task), task,
-				 sizeof(dsp_task_status_s) *
-					 query_task.return_num)) {
+										 sizeof(dsp_task_status_s) * query_task.return_num)) {
 			ret = -EINVAL;
 			dsp_err("copy to user task status err.\n");
 		}
@@ -724,8 +682,7 @@ static long dsp_query_task(struct file *flip, dsp_ioctl_query_s *query_task)
 	struct khandle *task_handle = NULL;
 
 	query_task->finish = -ENODATA;
-	task_handle = find_kernel_handle(&dsp_file->h, query_task->task_handle,
-					 DSP_REQ_HANDLE_MAGIC);
+	task_handle = find_kernel_handle(&dsp_file->h, query_task->task_handle, DSP_REQ_HANDLE_MAGIC);
 	if (task_handle == NULL) {
 		return -ENODATA;
 	}
@@ -736,9 +693,8 @@ static long dsp_query_task(struct file *flip, dsp_ioctl_query_s *query_task)
 	if (list_empty_careful(&dsp_file->async_ll_complete)) {
 		spin_unlock_irqrestore(&dsp_file->async_ll_lock, flags);
 		if (query_task->block != 0) {
-			ret = wait_event_interruptible(
-				dsp_file->async_ll_wq,
-				!list_empty(&dsp_file->async_ll_complete));
+			ret = wait_event_interruptible(dsp_file->async_ll_wq,
+								!list_empty(&dsp_file->async_ll_complete));
 		} else {
 			kernel_handle_decref(task_handle);
 			return 0;
@@ -799,8 +755,7 @@ static void dsp_release_user_completed_task(struct dsp_file *dsp_file)
 	unsigned long flags;
 	struct list_head *tmp_list;
 	spin_lock_irqsave(&dsp_file->async_ll_lock, flags);
-	list_for_each_entry_safe(async_task, tmp, &dsp_file->async_ll_complete,
-				 async_ll) {
+	list_for_each_entry_safe(async_task, tmp, &dsp_file->async_ll_complete, async_ll) {
 		spin_unlock_irqrestore(&dsp_file->async_ll_lock, flags);
 		list_del_init(&async_task->async_ll);
 		kernel_handle_decref(&async_task->handle);
@@ -822,8 +777,7 @@ static long dsp_ioctl_unload_op(struct file *flip, void *arg)
 	dsp_debug("", __func__, __LINE__);
 	user = dsp_find_user_by_fd(dsp_file, fd);
 	if (user == NULL) {
-		dsp_debug("%s, %d, cannot find user fd=%d.\n", __func__,
-			  __LINE__, fd);
+		dsp_debug("%s, %d, cannot find user fd=%d.\n", __func__, __LINE__, fd);
 		return -EINVAL;
 	}
 	kernel_handle_release_family(&user->h);
@@ -840,8 +794,7 @@ static void dsp_dma_buf_release(struct khandle *h)
 	struct es_dsp *dsp = dsp_file->dsp;
 
 	dsp_put_dma_addr(dsp, &entry->buf);
-	dsp_debug("%s, %d, fd=0x%x, iova=0x%x.\n", __func__, __LINE__,
-		  entry->buf.fd, entry->buf.dma_addr);
+	dsp_debug("%s, %d, fd=0x%x, iova=0x%x.\n", __func__, __LINE__, entry->buf.fd, entry->buf.dma_addr);
 	kfree(entry);
 }
 
@@ -865,8 +818,7 @@ static long dsp_ioctl_prepare_dma(struct file *flip, dsp_ioctl_pre_dma_s *arg)
 		khandle = find_kernel_handle(&dsp_file->h, entry->handle.fd,
 					     DSP_DMABUF_HANDLE_MAGIC);
 		BUG_ON(khandle == NULL);
-		dsp_debug("%s, %d, find %d entry, handle fd=%d.\n", __func__,
-			  __LINE__, entry->buf.fd, entry->handle.fd);
+		dsp_debug("%s, %d, find %d entry, handle fd=%d.\n", __func__, __LINE__, entry->buf.fd, entry->handle.fd);
 		entry->count++;
 		kernel_handle_decref(khandle);
 		mutex_unlock(&dsp_file->xrray_lock);
@@ -875,8 +827,7 @@ static long dsp_ioctl_prepare_dma(struct file *flip, dsp_ioctl_pre_dma_s *arg)
 
 	entry = kzalloc(sizeof(struct dsp_dma_buf_ex), GFP_KERNEL);
 	if (!entry) {
-		dsp_err("%s, %d, alloc memory for device dma buf err.\n",
-			__func__, __LINE__);
+		dsp_err("%s, %d, alloc memory for device dma buf err.\n", __func__, __LINE__);
 		mutex_unlock(&dsp_file->xrray_lock);
 		return -ENOMEM;
 	}
@@ -884,15 +835,13 @@ static long dsp_ioctl_prepare_dma(struct file *flip, dsp_ioctl_pre_dma_s *arg)
 	ret = init_kernel_handle(&entry->handle, dsp_dma_buf_release,
 				 DSP_DMABUF_HANDLE_MAGIC, &dsp_file->h);
 	if (ret) {
-		dsp_err("%s, %d, init kernel handle err.\n", __func__,
-			__LINE__);
+		dsp_err("%s, %d, init kernel handle err.\n", __func__, __LINE__);
 		kfree(entry);
 		mutex_unlock(&dsp_file->xrray_lock);
 		return ret;
 	}
 
-	dsp_debug("%s, %d, handle_fd=%d.\n", __func__, __LINE__,
-		  entry->handle.fd);
+	dsp_debug("%s, %d, handle_fd=%d.\n", __func__, __LINE__, entry->handle.fd);
 	addr = dsp_get_dma_addr(dsp, (int)buf.desc.memFd, &entry->buf);
 	if (addr == 0) {
 		dsp_err("%s, %d, addr err.\n", __func__, __LINE__);
@@ -934,8 +883,7 @@ static long dsp_ioctl_unprepare_dma(struct file *flip, u32 *arg)
 	mutex_lock(&dsp_file->xrray_lock);
 	entry = xa_load(&dsp_file->buf_xrray, fd);
 	if (!entry) {
-		dsp_err("%s,%d, cannot find %d dmabuf entry.\n", __func__,
-			__LINE__, fd);
+		dsp_err("%s,%d, cannot find %d dmabuf entry.\n", __func__, __LINE__, fd);
 		mutex_unlock(&dsp_file->xrray_lock);
 		return -EINVAL;
 	}
@@ -945,10 +893,8 @@ static long dsp_ioctl_unprepare_dma(struct file *flip, u32 *arg)
 		mutex_unlock(&dsp_file->xrray_lock);
 		return 0;
 	}
-	dsp_debug("%s, %d, entry->fd=%d.\n\n", __func__, __LINE__,
-		  entry->buf.fd);
-	khandle = find_kernel_handle(&dsp_file->h, entry->handle.fd,
-				     DSP_DMABUF_HANDLE_MAGIC);
+	dsp_debug("%s, %d, entry->fd=%d.\n\n", __func__, __LINE__, entry->buf.fd);
+	khandle = find_kernel_handle(&dsp_file->h, entry->handle.fd, DSP_DMABUF_HANDLE_MAGIC);
 
 	BUG_ON(!khandle);
 	xa_erase(&dsp_file->buf_xrray, entry->buf.fd);
@@ -975,11 +921,7 @@ static long dsp_ioctl_get_perf_data(struct file *flip, dsp_kmd_perf_t *data)
 {
 	int ret = 0;
 
-	struct dsp_file *dsp_file = flip->private_data;
-	struct es_dsp *dsp = dsp_file->dsp;
-
-	if (copy_to_user(data, dsp->op_perf,
-			 sizeof(dsp_kmd_perf_t) * MAX_DSP_TASKS)) {
+	if (copy_to_user(data, 0, sizeof(dsp_kmd_perf_t) * MAX_DSP_TASKS)) {
 		dsp_err("copy perf data to user err.\n");
 		ret = -EINVAL;
 	}
@@ -995,31 +937,10 @@ static long dsp_ioctl_get_fw_perf_data(struct file *flip, dsp_fw_perf_t *data)
 {
 	int ret = 0;
 
-	struct dsp_file *dsp_file = flip->private_data;
-	struct es_dsp *dsp = dsp_file->dsp;
-	es_dsp_perf_info perf_info;
-
-	get_dsp_perf_info(&perf_info, 0, 0);
-
-	dsp->op_fw_perf[0].Die = 0;
-	dsp->op_fw_perf[0].CoreId = perf_info.core_id;
-	dsp->op_fw_perf[0].OpIndex = perf_info.op_index;
-	dsp->op_fw_perf[0].OpType = perf_info.op_type;
-	// dsp->op_fw_perf[0].OpName = "";
-	dsp->op_fw_perf[0].OpStartCycle = perf_info.flat1_start_time;
-	dsp->op_fw_perf[0].OpPrepareStartCycle = perf_info.prepare_start_time;
-	dsp->op_fw_perf[0].OpPrepareEndCycle = perf_info.prepare_end_time;
-	dsp->op_fw_perf[0].OpEvalStartCycle = perf_info.eval_start_time;
-	dsp->op_fw_perf[0].OpEvalEndCycle = perf_info.eval_end_time;
-	dsp->op_fw_perf[0].OpNotifyStartCycle = perf_info.notify_start_time;
-	dsp->op_fw_perf[0].OpEndCycle = perf_info.flat1_end_time;
-
-	if (copy_to_user(data, dsp->op_fw_perf,
-			 sizeof(dsp_fw_perf_t) * MAX_DSP_TASKS)) {
+	if (copy_to_user(data, 0, sizeof(dsp_fw_perf_t) * MAX_DSP_TASKS)) {
 		dsp_err("copy perf data to user err.\n");
 		ret = -EINVAL;
 	}
-	dsp_debug("get dsp%u hardware perf data done.\n", perf_info.op_index);
 
 	return ret;
 }
@@ -1032,9 +953,9 @@ static long dsp_ioctl_get_cur_op_perf_data(struct file *flip, dsp_fw_perf_t *dat
 	struct es_dsp *dsp = dsp_file->dsp;
 	es_dsp_perf_info perf_info;
 
-	get_dsp_perf_info(&perf_info, 0, 0);
+	get_dsp_perf_info(&perf_info, dsp->numa_id, dsp->process_id);
 
-	dsp->op_cur_perf.Die = 0;
+	dsp->op_cur_perf.Die = dsp->numa_id;
 	dsp->op_cur_perf.CoreId = perf_info.core_id;
 	dsp->op_cur_perf.OpIndex = perf_info.op_index;
 	dsp->op_cur_perf.OpType = perf_info.op_type;
@@ -1066,7 +987,6 @@ static long dsp_ioctl_multi_tasks_submit(struct file *flip,
 	struct dsp_user_req_async **user_req;
 	int i, ret;
 	unsigned long flags;
-	int vm_vmalloc = 0;
 
 	if (copy_from_user(&req, arg, sizeof(dsp_ioctl_task_s))) {
 		dsp_err("%s, %d, copy_from_user err.\n", __func__, __LINE__);
@@ -1079,29 +999,19 @@ static long dsp_ioctl_multi_tasks_submit(struct file *flip,
 		return ret;
 	}
 	if (req.task_num <= 0) {
-		dsp_err("%s, %d, task num below zero, err,\n", __func__,
-			__LINE__);
+		dsp_err("%s, %d, task num below zero, err,\n", __func__, __LINE__);
 		return -EINVAL;
 	}
-	tasks = kzalloc(req.task_num * (sizeof(dsp_ioctl_task_s) +
-					sizeof(struct dsp_user_req_async *)),
-			GFP_KERNEL | __GFP_NOWARN);
-	if (tasks == NULL) {
-		tasks = vzalloc(req.task_num * (sizeof(dsp_ioctl_task_s) +
-					sizeof(struct dsp_user_req_async *)));
-		vm_vmalloc = 1;
-
-	}
 
+	tasks = vzalloc(req.task_num * (sizeof(dsp_ioctl_task_s) +
+				sizeof(struct dsp_user_req_async *)));
 	if (tasks == NULL) {
 		dsp_err("", __func__, __LINE__);
 		return -ENOMEM;
 	}
 
-	if (copy_from_user(tasks, arg,
-			   req.task_num * sizeof(dsp_ioctl_task_s))) {
-		dsp_err("%s, %d, copy_from_user for multi tasks err.\n",
-			__func__, __LINE__);
+	if (copy_from_user(tasks, arg, req.task_num * sizeof(dsp_ioctl_task_s))) {
+		dsp_err("%s, %d, copy_from_user for multi tasks err.\n", __func__, __LINE__);
 		ret = -EINVAL;
 		goto err0;
 	}
@@ -1115,8 +1025,7 @@ static long dsp_ioctl_multi_tasks_submit(struct file *flip,
 	for (i = 0; i < req.task_num; i++) {
 		user_req[i] = dsp_set_task_info(dsp_file, &tasks[i], false);
 		if (user_req[i] == NULL) {
-			dsp_err("%s, %d, and ,i = %d.\n", __func__, __LINE__,
-				i);
+			dsp_err("%s, %d, and ,i = %d.\n", __func__, __LINE__, i);
 			goto free_task;
 		}
 		tasks[i].task.taskHandle = user_req[i]->handle.fd;
@@ -1138,11 +1047,8 @@ static long dsp_ioctl_multi_tasks_submit(struct file *flip,
 		dsp_err("copy to user err.\n");
 		ret = -EINVAL;
 	}
-	if (vm_vmalloc) {
-		vfree(tasks);
-	} else {
-		kfree(tasks);
-	}
+	vfree(tasks);
+
 	return 0;
 
 free_task:
@@ -1153,11 +1059,8 @@ static long dsp_ioctl_multi_tasks_submit(struct file *flip,
 
 	module_put(THIS_MODULE);
 err0:
-	if (vm_vmalloc) {
-		vfree(tasks);
-	} else {
-		kfree(tasks);
-	}
+	vfree(tasks);
+
 	return ret;
 }
 static long dsp_ioctl(struct file *flip, unsigned int cmd, unsigned long arg)
diff --git a/drivers/soc/eswin/ai_driver/dsp/dsp_main.h b/drivers/soc/eswin/ai_driver/dsp/dsp_main.h
index b225e2287ff4..ac6e04357adf 100644
--- a/drivers/soc/eswin/ai_driver/dsp/dsp_main.h
+++ b/drivers/soc/eswin/ai_driver/dsp/dsp_main.h
@@ -188,8 +188,6 @@ struct es_dsp {
 
 	u32 perf_enable;
 	int op_idx;
-	dsp_kmd_perf_t op_perf[MAX_DSP_TASKS];
-	dsp_fw_perf_t op_fw_perf[MAX_DSP_TASKS];
 	dsp_fw_perf_t op_cur_perf;
 	atomic_t dmabuf_mapped_cnt;
 #if defined(CONFIG_PM_DEVFREQ)
diff --git a/drivers/soc/eswin/ai_driver/dsp/dsp_pool.c b/drivers/soc/eswin/ai_driver/dsp/dsp_pool.c
index 5da8214d8605..f578d1731999 100644
--- a/drivers/soc/eswin/ai_driver/dsp/dsp_pool.c
+++ b/drivers/soc/eswin/ai_driver/dsp/dsp_pool.c
@@ -110,7 +110,7 @@ struct dsp_pool *dsp_pool_create(struct device *dev, size_t size,
 void *dsp_pool_alloc(struct dsp_pool *pool, gfp_t mem_flags, dma_addr_t *handle)
 {
 	unsigned long flags;
-	struct dsp_pool_page *page;
+	struct dsp_pool_page *page = NULL;
 	void *retval;
 	size_t offset;
 
@@ -139,7 +139,7 @@ void *dsp_pool_alloc(struct dsp_pool *pool, gfp_t mem_flags, dma_addr_t *handle)
 static struct dsp_pool_page *dsp_pool_find_page(struct dsp_pool *pool,
 						dma_addr_t dma)
 {
-	struct dsp_pool_page *page;
+	struct dsp_pool_page *page =  NULL;
 	list_for_each_entry(page, &pool->page_list, page_list) {
 		if (dma < page->dma)
 			continue;
@@ -179,7 +179,7 @@ static void dsp_pool_free_page(struct dsp_pool *pool,
 
 void dsp_pool_destroy(struct dsp_pool *pool)
 {
-	struct dsp_pool_page *page, *tmp;
+	struct dsp_pool_page *page = NULL, *tmp = NULL;
 	if (!pool) {
 		return;
 	}
diff --git a/drivers/soc/eswin/ai_driver/npu/dla_buffer.c b/drivers/soc/eswin/ai_driver/npu/dla_buffer.c
index 8aea165e53ec..1db428181eed 100644
--- a/drivers/soc/eswin/ai_driver/npu/dla_buffer.c
+++ b/drivers/soc/eswin/ai_driver/npu/dla_buffer.c
@@ -282,8 +282,7 @@ int dla_import_dmabuf_from_model(struct user_model *model)
 	nvdla_dev = model->nvdla_dev;
 
 	dla_detail("num_address=%d\n", num_address);
-	bobjs = kzalloc(num_address * sizeof(struct dla_buffer_object),
-			GFP_KERNEL);
+	bobjs = vzalloc(num_address * sizeof(struct dla_buffer_object));
 	if (bobjs == NULL) {
 		dla_error("err:bobjs is NULL,no memory!\n");
 		return -ENOMEM;
@@ -331,7 +330,7 @@ int dla_import_dmabuf_from_model(struct user_model *model)
 		}
 	}
 
-	kfree(bobjs);
+	vfree(bobjs);
 
 	return ret;
 }
@@ -358,7 +357,7 @@ int dla_detach_dmabuf_from_model(struct user_model *model)
 		}
 	}
 
-	kfree(bobjs);
+	vfree(bobjs);
 
 	return ret;
 }
diff --git a/drivers/soc/eswin/ai_driver/npu/dsp.c b/drivers/soc/eswin/ai_driver/npu/dsp.c
index 15e43027b1a5..a796dc577520 100644
--- a/drivers/soc/eswin/ai_driver/npu/dsp.c
+++ b/drivers/soc/eswin/ai_driver/npu/dsp.c
@@ -73,8 +73,7 @@ static void npu_put_dsp_dmabuf(struct file *file, struct dsp_dma_buf **dma_buf)
 		dla_error("%s, %d, dsp file is null.\n", __func__, __LINE__);
 		goto err;
 	}
-	dla_debug("%s, %d, file count=%ld.\n", __func__, __LINE__,
-		  file_count(file));
+	dla_debug("%s, %d, file count=%ld.\n", __func__, __LINE__, file_count(file));
 	dsp_unmap_dmabuf(dsp_file, dma_buf, 1);
 err:
 	return;
@@ -94,8 +93,7 @@ static u32 npu_get_dsp_ddr(struct win_executor *executor, int devid, int index)
 
 	memfd = task->bobjs[index].fd;
 	if (memfd < 0) {
-		dla_error("%s, %d, index=%d, memfd is not invalid.\n", __func__,
-			  __LINE__, index);
+		dla_error("%s, %d, index=%d, memfd is not invalid.\n", __func__, __LINE__, index);
 		return 0;
 	}
 	mutex_lock(&executor->xrray_lock[devid]);
@@ -108,9 +106,8 @@ static u32 npu_get_dsp_ddr(struct win_executor *executor, int devid, int index)
 	dmabuf = npu_get_dsp_dmabuf(executor->dsp_file[devid], memfd);
 	if (dmabuf == NULL) {
 		mutex_unlock(&executor->xrray_lock[devid]);
-		dla_error(
-			"%s, %d, cannot get index=%d, memfd=%d dmabuf, error.\n",
-			__func__, __LINE__, index, memfd);
+		dla_error("%s, %d, cannot get index=%d, memfd=%d dmabuf, error.\n",
+				  __func__, __LINE__, index, memfd);
 		return 0;
 	}
 
@@ -155,30 +152,25 @@ static int dsp_tensor_unfold(struct win_executor *executor, int op_idx,
 
 	dsp_tensor = &tensor_set[idx];
 	dla_debug("%s, %d, op_idx=%d, idx=%d, dsp_tensor=0x%px.\n", __func__,
-		  __LINE__, op_idx, idx, dsp_tensor);
+			  __LINE__, op_idx, idx, dsp_tensor);
 
 	if (tensor_set != NULL && dsp_tensor != NULL &&
 	    dsp_tensor->have_unfold) {
 		dsp->npu_info.current_op_idx = op_idx;
-		ret = load_operator(executor->engine->dsp_dev[dsp_dev_idx],
-				    NULL, dsp_op->operator_name,
-				    &dsp_tensor->handle);
+		ret = load_operator(executor->engine->dsp_dev[dsp_dev_idx], NULL,
+							dsp_op->operator_name, &dsp_tensor->handle);
 		if (ret != 0) {
-			dla_error(
-				"err:dsp load_operator failed,ret:%d dsp_dev_idx:%d idx:%d\n",
-				ret, dsp_dev_idx, idx);
+			dla_error("err:dsp load_operator failed,ret:%d dsp_dev_idx:%d idx:%d\n",
+					  ret, dsp_dev_idx, idx);
 			return -1;
 		}
-		dla_debug(
-			"%s, %d, op_idx=%d, dsp_tensor=0x%px, handle=0x%llx.\n",
-			__func__, __LINE__, op_idx, dsp_tensor,
-			dsp_tensor->handle);
+		dla_debug("%s, %d, op_idx=%d, dsp_tensor=0x%px, handle=0x%llx.\n",
+				  __func__, __LINE__, op_idx, dsp_tensor, dsp_tensor->handle);
 		return 0;
 	}
 
 	flat1_size = (dsp_op->buffer_cnt_cfg + dsp_op->buffer_cnt_input +
-		      dsp_op->buffer_cnt_output) *
-		     sizeof(es_dsp_buffer);
+				  dsp_op->buffer_cnt_output) * sizeof(es_dsp_buffer);
 	flat1_size += sizeof(struct es_dsp_flat1_desc);
 	flat1_size = round_up(flat1_size, 64);
 
@@ -189,27 +181,23 @@ static int dsp_tensor_unfold(struct win_executor *executor, int op_idx,
 
 	dsp_tensor = &tensor_set[idx];
 	dla_debug("%s, %d, op_type=%d.\n", __func__, __LINE__, op_type);
-	dla_debug("%s, %d, idx=%d, tensor_set=0x%px.\n", __func__, __LINE__,
-		  idx, tensor_set);
-	dla_debug("%s, %d, idx=%d, dsp_tensor=0x%px.\n", __func__, __LINE__,
-		  idx, dsp_tensor);
+	dla_debug("%s, %d, idx=%d, tensor_set=0x%px.\n", __func__, __LINE__, idx, tensor_set);
+	dla_debug("%s, %d, idx=%d, dsp_tensor=0x%px.\n", __func__, __LINE__, idx, dsp_tensor);
 	for (i = 0; i < dsp_op->buffer_cnt_input; i++) {
 		ret = read_input_address(executor, &surface->src_data[i],
 					 (void *)&dsp_tensor->src_base_addr[i],
 					 &dsp_tensor->src_is_io_tensor[i]);
 		if (ret != 0) {
-			dla_error("%s %d bad memory type %d\n", __func__,
-				  __LINE__, surface->src_data[i].type);
+			dla_error("%s %d bad memory type %d\n", __func__, __LINE__, surface->src_data[i].type);
 			return -1;
 		}
 		if (dsp_tensor->src_is_io_tensor[i] != invalid_tensor_idx) {
-			executor->dsp_all_inout
-				[dsp_dev_idx][dsp_tensor->src_is_io_tensor[i]]++;
+			executor->dsp_all_inout[dsp_dev_idx][dsp_tensor->src_is_io_tensor[i]]++;
 			executor->dsp_iobuf_cnt[dsp_dev_idx]++;
 		}
 		dla_debug("i=%d src_base_addr=0x%llx src_is_io_tensor=0x%x\n",
-			  i, dsp_tensor->src_base_addr[i],
-			  dsp_tensor->src_is_io_tensor[i]);
+				  i, dsp_tensor->src_base_addr[i],
+				  dsp_tensor->src_is_io_tensor[i]);
 	}
 
 	for (i = 0; i < dsp_op->buffer_cnt_output; i++) {
@@ -217,21 +205,18 @@ static int dsp_tensor_unfold(struct win_executor *executor, int op_idx,
 					 (void *)&dsp_tensor->dst_base_addr[i],
 					 &dsp_tensor->dst_is_io_tensor[i]);
 		if (ret != 0) {
-			dla_error("%s %d bad memory type %d\n", __func__,
-				  __LINE__, surface->dst_data[i].type);
+			dla_error("%s %d bad memory type %d\n", __func__, __LINE__, surface->dst_data[i].type);
 			return -1;
 		}
 		if (dsp_tensor->dst_is_io_tensor[i] != invalid_tensor_idx) {
-			executor->dsp_all_inout
-				[dsp_dev_idx][dsp_tensor->dst_is_io_tensor[i]]++;
+			executor->dsp_all_inout[dsp_dev_idx][dsp_tensor->dst_is_io_tensor[i]]++;
 			executor->dsp_iobuf_cnt[dsp_dev_idx]++;
 		}
 
-		dla_debug(
-			"i=%d op_idx=%d, type=%d, dst_base_addr=0x%llx dst_is_io_tensor=0x%x\n",
-			i, op_idx, surface->dst_data[i].type,
-			dsp_tensor->dst_base_addr[i],
-			dsp_tensor->dst_is_io_tensor[i]);
+		dla_debug("i=%d op_idx=%d, type=%d, dst_base_addr=0x%llx dst_is_io_tensor=0x%x\n",
+				  i, op_idx, surface->dst_data[i].type,
+				  dsp_tensor->dst_base_addr[i],
+				  dsp_tensor->dst_is_io_tensor[i]);
 	}
 
 	dsp_tensor->have_unfold = 1;
@@ -304,35 +289,27 @@ static int processor_dsp_program(struct win_executor *executor, int idx,
 
 	flat1_desc = (struct es_dsp_flat1_desc *)dsp_tensor->flat1_vaddr;
 
-	dla_debug("%s, %d, flat1_desc=0x%px.\n", __func__, __LINE__,
-		  flat1_desc);
-	dla_debug("%s, %d, cnt_cfg=%d.\n", __func__, __LINE__,
-		  dsp_op->buffer_cnt_cfg);
-	dla_debug("%s, %d, cnt_input=%d.\n", __func__, __LINE__,
-		  dsp_op->buffer_cnt_input);
-	dla_debug("%s, %d, cnt_output=%d.\n", __func__, __LINE__,
-		  dsp_op->buffer_cnt_output);
-	flat1_desc->num_buffer = dsp_op->buffer_cnt_cfg +
-				 dsp_op->buffer_cnt_input +
-				 dsp_op->buffer_cnt_output;
-
-	dla_debug("%s, %d op_handle=0x%llx.\n", __func__, __LINE__,
-		  dsp_tensor->handle);
+	dla_debug("%s, %d, flat1_desc=0x%px.\n", __func__, __LINE__, flat1_desc);
+	dla_debug("%s, %d, cnt_cfg=%d.\n", __func__, __LINE__, dsp_op->buffer_cnt_cfg);
+	dla_debug("%s, %d, cnt_input=%d.\n", __func__, __LINE__, dsp_op->buffer_cnt_input);
+	dla_debug("%s, %d, cnt_output=%d.\n", __func__, __LINE__, dsp_op->buffer_cnt_output);
+	flat1_desc->num_buffer = dsp_op->buffer_cnt_cfg + dsp_op->buffer_cnt_input +
+							 dsp_op->buffer_cnt_output;
+
+	dla_debug("%s, %d op_handle=0x%llx.\n", __func__, __LINE__, dsp_tensor->handle);
 	dsp_set_flat_func(flat1_desc, dsp_tensor->handle);
 
 	flat1_desc->input_index = dsp_op->buffer_cnt_cfg;
-	flat1_desc->output_index =
-		dsp_op->buffer_cnt_cfg + dsp_op->buffer_cnt_input;
+	flat1_desc->output_index = dsp_op->buffer_cnt_cfg + dsp_op->buffer_cnt_input;
 	dsp_buffer = flat1_desc->buffers;
 
 	for (i = 0; i < dsp_op->buffer_cnt_cfg; i++) {
-		dsp_buffer[i].addr = dsp_tensor->data_dma_addr +
-				     param_data_loac_offset + offset +
-				     param_addr_offset;
+		dsp_buffer[i].addr = dsp_tensor->data_dma_addr + param_data_loac_offset +
+							 offset + param_addr_offset;
 		dsp_buffer[i].size = dsp_op->buffer_size[i];
 		param_addr_offset += dsp_op->buffer_size[i];
 		dla_debug("%s, %d cfg=%d , addr=0x%x, size=0x%x.\n", __func__,
-			  __LINE__, i, dsp_buffer[i].addr, dsp_buffer[i].size);
+				  __LINE__, i, dsp_buffer[i].addr, dsp_buffer[i].size);
 	}
 
 	dsp_buf_idx = dsp_op->buffer_cnt_cfg;
@@ -340,35 +317,25 @@ static int processor_dsp_program(struct win_executor *executor, int idx,
 	for (i = 0; i < dsp_op->buffer_cnt_input; i++) {
 		if (dsp_tensor->src_is_io_tensor[i] == invalid_tensor_idx) {
 			if (surface->src_data[i].type == DLA_MEM_MC) {
-				dsp_addr = npu_get_dsp_ddr( executor, dev_id,
-						surface->src_data[i].address);
+				dsp_addr = npu_get_dsp_ddr( executor, dev_id, surface->src_data[i].address);
 				if(!dsp_addr) {
 					return -1;
 				}
-				dsp_buffer[dsp_buf_idx + i].addr = dsp_addr +
-					surface->src_data[i].offset;
+				dsp_buffer[dsp_buf_idx + i].addr = dsp_addr + surface->src_data[i].offset;
 			} else if (surface->src_data[i].type == DLA_MEM_CV) {
-				dsp_buffer[dsp_buf_idx + i]
-					.addr = dsp_get_sram_iova_by_addr(
-					executor->engine->dsp_dev[dev_id],
-					dsp_tensor->src_base_addr[i]);
-				dla_debug("get dsp sram iova = 0x%x.\n",
-					  dsp_buffer[dsp_buf_idx + i].addr);
+				dsp_buffer[dsp_buf_idx + i].addr =
+						dsp_get_sram_iova_by_addr(executor->engine->dsp_dev[dev_id],
+												  dsp_tensor->src_base_addr[i]);
+				dla_debug("get dsp sram iova = 0x%x.\n", dsp_buffer[dsp_buf_idx + i].addr);
 			}
 		} else {
-			dsp_buffer[dsp_buf_idx + i].addr =
-				surface->src_data[i].offset;
-			io_cnt =
-				executor->dsp_io[dev_id]
-						[dsp_tensor->src_is_io_tensor[i]]
-							.io_cnt;
-			executor->dsp_io[dev_id][dsp_tensor->src_is_io_tensor[i]]
-				.virt[io_cnt] =
-				(u64)&dsp_buffer[dsp_buf_idx + i].addr;
-			executor->dsp_io[dev_id][dsp_tensor->src_is_io_tensor[i]]
-				.offset[io_cnt] = surface->src_data[i].offset;
-			executor->dsp_io[dev_id][dsp_tensor->src_is_io_tensor[i]]
-				.io_cnt++;
+			dsp_buffer[dsp_buf_idx + i].addr = surface->src_data[i].offset;
+			io_cnt = executor->dsp_io[dev_id][dsp_tensor->src_is_io_tensor[i]].io_cnt;
+			executor->dsp_io[dev_id][dsp_tensor->src_is_io_tensor[i]].virt[io_cnt] =
+								(u64)&dsp_buffer[dsp_buf_idx + i].addr;
+			executor->dsp_io[dev_id][dsp_tensor->src_is_io_tensor[i]].offset[io_cnt] =
+								surface->src_data[i].offset;
+			executor->dsp_io[dev_id][dsp_tensor->src_is_io_tensor[i]].io_cnt++;
 		}
 		dsp_buffer[dsp_buf_idx + i].size = surface->src_data[i].size;
 	}
@@ -377,61 +344,39 @@ static int processor_dsp_program(struct win_executor *executor, int idx,
 	for (i = 0; i < dsp_op->buffer_cnt_output; i++) {
 		if (dsp_tensor->dst_is_io_tensor[i] == invalid_tensor_idx) {
 			if (surface->dst_data[i].type == DLA_MEM_MC) {
-				dla_debug(
-					"%s, %d, output dst mc, address index=%d.\n",
-					__func__, __LINE__,
-					surface->dst_data[i].address);
-				dsp_addr = npu_get_dsp_ddr( executor, dev_id,
-						surface->dst_data[i].address);
+				dla_debug("%s, %d, output dst mc, address index=%d.\n", __func__, __LINE__,
+						  surface->dst_data[i].address);
+				dsp_addr = npu_get_dsp_ddr( executor, dev_id, surface->dst_data[i].address);
 				if(!dsp_addr) {
 					return -1;
 				}
-				dsp_buffer[dsp_buf_idx + i].addr = dsp_addr +
-					surface->dst_data[i].offset;
-				dla_debug("%s, %d, i=%d, addr=0x%x.\n",
-					  __func__, __LINE__, i,
-					  dsp_buffer[dsp_buf_idx + i].addr);
+				dsp_buffer[dsp_buf_idx + i].addr = dsp_addr + surface->dst_data[i].offset;
+				dla_debug("%s, %d, i=%d, addr=0x%x.\n", __func__, __LINE__, i,
+						  dsp_buffer[dsp_buf_idx + i].addr);
 			} else if (surface->dst_data[i].type == DLA_MEM_CV) {
-				dsp_buffer[dsp_buf_idx + i]
-					.addr = dsp_get_sram_iova_by_addr(
-					executor->engine->dsp_dev[dev_id],
-					dsp_tensor->dst_base_addr[i]);
-				dla_debug(
-					"%s, %d, cv. output offset=0x%x, addr=0x%x.\n",
-					__func__, __LINE__,
-					surface->dst_data[i].offset,
-					dsp_buffer[dsp_buf_idx + i].addr);
+				dsp_buffer[dsp_buf_idx + i].addr =
+							 dsp_get_sram_iova_by_addr(executor->engine->dsp_dev[dev_id],
+													   dsp_tensor->dst_base_addr[i]);
+				dla_debug("%s, %d, cv. output offset=0x%x, addr=0x%x.\n", __func__, __LINE__,
+						  surface->dst_data[i].offset, dsp_buffer[dsp_buf_idx + i].addr);
 			}
-			dla_debug(
-				"%s, %d, i=%d, outidx=%d, buffer addr=0x%x, offset=0x%x. size=0x%x.\n",
-				__func__, __LINE__, i, dsp_buf_idx + i,
-				dsp_buffer[dsp_buf_idx + i].addr,
-				surface->dst_data[i].offset,
-				dsp_buffer[dsp_buf_idx + i].size);
+			dla_debug("%s, %d, i=%d, outidx=%d, buffer addr=0x%x, offset=0x%x. size=0x%x.\n",
+					  __func__, __LINE__, i, dsp_buf_idx + i,
+					  dsp_buffer[dsp_buf_idx + i].addr, surface->dst_data[i].offset,
+					  dsp_buffer[dsp_buf_idx + i].size);
 		} else {
-			dsp_buffer[dsp_buf_idx + i].addr =
-				surface->dst_data[i].offset;
-			io_cnt =
-				executor->dsp_io[dev_id]
-						[dsp_tensor->dst_is_io_tensor[i]]
-							.io_cnt;
-			executor->dsp_io[dev_id][dsp_tensor->dst_is_io_tensor[i]]
-				.virt[io_cnt] =
-				(u64)&dsp_buffer[dsp_buf_idx + i].addr;
-			executor->dsp_io[dev_id][dsp_tensor->dst_is_io_tensor[i]]
-				.offset[io_cnt] = surface->dst_data[i].offset;
-			dla_debug(
-				"%s, %d, io_cnt=%d, dsp_io=%d, dst=0x%llx.\n",
-				__func__, __LINE__, io_cnt,
-				dsp_tensor->dst_is_io_tensor[i],
-				executor->dsp_io[dev_id]
-						[dsp_tensor->dst_is_io_tensor[i]]
-							.virt[io_cnt]);
+			dsp_buffer[dsp_buf_idx + i].addr = surface->dst_data[i].offset;
+			io_cnt = executor->dsp_io[dev_id][dsp_tensor->dst_is_io_tensor[i]].io_cnt;
+			executor->dsp_io[dev_id][dsp_tensor->dst_is_io_tensor[i]].virt[io_cnt] =
+					(u64)&dsp_buffer[dsp_buf_idx + i].addr;
+			executor->dsp_io[dev_id][dsp_tensor->dst_is_io_tensor[i]].offset[io_cnt] =
+					surface->dst_data[i].offset;
+			dla_debug("%s, %d, io_cnt=%d, dsp_io=%d, dst=0x%llx.\n",
+					__func__, __LINE__, io_cnt, dsp_tensor->dst_is_io_tensor[i],
+					executor->dsp_io[dev_id][dsp_tensor->dst_is_io_tensor[i]].virt[io_cnt]);
 			dla_debug("%s, %d, output %d, offset=0x%x.\n", __func__,
-				  __LINE__, dsp_buf_idx + i,
-				  dsp_buffer[dsp_buf_idx + i].addr);
-			executor->dsp_io[dev_id][dsp_tensor->dst_is_io_tensor[i]]
-				.io_cnt++;
+					__LINE__, dsp_buf_idx + i, dsp_buffer[dsp_buf_idx + i].addr);
+					executor->dsp_io[dev_id][dsp_tensor->dst_is_io_tensor[i]].io_cnt++;
 		}
 		dsp_buffer[dsp_buf_idx + i].size = surface->dst_data[i].size;
 	}
@@ -443,8 +388,7 @@ static int processor_dsp_program(struct win_executor *executor, int idx,
 	h2d_msg.sync_cache = 0;
 
 	h2d_msg.iova_ptr = dsp_tensor->flat1_dma_addr;
-	memcpy((void *)&dsp->npu_info.dsp_eval_param, (void *)&h2d_msg,
-	       sizeof(h2d_msg));
+	memcpy((void *)&dsp->npu_info.dsp_eval_param, (void *)&h2d_msg, sizeof(h2d_msg));
 	return 0;
 }
 
@@ -454,8 +398,7 @@ int dsp0_prepare_prog_data(struct win_executor *executor, int rdma, int idx,
 {
 	int ret;
 	dla_debug("%s, %d, op_idx=%d.\n", __func__, __LINE__, op_idx);
-	ret = processor_dsp_program(executor, idx, op_idx, op_desc, surf_desc,
-				    DLA_KMD_OP_DSP_0);
+	ret = processor_dsp_program(executor, idx, op_idx, op_desc, surf_desc, DLA_KMD_OP_DSP_0);
 	return ret;
 }
 
@@ -465,8 +408,7 @@ int dsp1_prepare_prog_data(struct win_executor *executor, int rdma, int idx,
 {
 	int ret;
 
-	ret = processor_dsp_program(executor, idx, op_idx, op_desc, surf_desc,
-				    DLA_KMD_OP_DSP_1);
+	ret = processor_dsp_program(executor, idx, op_idx, op_desc, surf_desc, DLA_KMD_OP_DSP_1);
 	return ret;
 }
 
@@ -476,8 +418,7 @@ int dsp2_prepare_prog_data(struct win_executor *executor, int rdma, int idx,
 {
 	int ret;
 
-	ret = processor_dsp_program(executor, idx, op_idx, op_desc, surf_desc,
-				    DLA_KMD_OP_DSP_2);
+	ret = processor_dsp_program(executor, idx, op_idx, op_desc, surf_desc, DLA_KMD_OP_DSP_2);
 	return ret;
 }
 
@@ -487,8 +428,7 @@ int dsp3_prepare_prog_data(struct win_executor *executor, int rdma, int idx,
 {
 	int ret;
 
-	ret = processor_dsp_program(executor, idx, op_idx, op_desc, surf_desc,
-				    DLA_KMD_OP_DSP_3);
+	ret = processor_dsp_program(executor, idx, op_idx, op_desc, surf_desc, DLA_KMD_OP_DSP_3);
 	return ret;
 }
 
@@ -507,17 +447,15 @@ static int npu_acquire_dsp_file(struct win_executor *executor)
 		filefd = executor->dsp_fd[i];
 		file = fget(filefd);
 		if (file == NULL) {
-			dla_error(
-				"%s, %d, cannot find dsp%d dsp_file fd=%d file, error\n",
-				__func__, __LINE__, i, filefd);
+			dla_error("%s, %d, cannot find dsp%d dsp_file fd=%d file, error\n",
+					  __func__, __LINE__, i, filefd);
 			executor->dsp_file[i] = NULL;
 			continue;
 		}
 
 		dsp_file = file->private_data;
 		if (dsp_file == NULL) {
-			dla_error("%s, %d, dsp file(%d) is null.\n", __func__,
-				  __LINE__, filefd);
+			dla_error("%s, %d, dsp file(%d) is null.\n", __func__, __LINE__, filefd);
 			executor->dsp_file[i] = NULL;
 			fput(file);
 			continue;
@@ -570,26 +508,20 @@ int resolve_dsp_data(struct win_executor *executor)
 		    (op_type == DLA_KMD_OP_DSP_2) ||
 		    (op_type == DLA_KMD_OP_DSP_3)) {
 			tensor_set = executor->tensor_set[op_type];
-			dsp_tensor =
-				&tensor_set[dsp_tensor_idx[op_type -
-							   DLA_KMD_OP_DSP_0]];
+			dsp_tensor = &tensor_set[dsp_tensor_idx[op_type - DLA_KMD_OP_DSP_0]];
 			pcer_interface = executor->engine->processors[op_type];
 
 			j = dsp_tensor_idx[op_type - DLA_KMD_OP_DSP_0];
-			size = pcer_interface->tensor_unfold(
-				executor, op_idx, &task->op_desc[op_idx],
-				&task->surface_desc[op_idx], tensor_set,
-				j);  //get dsp flat1 len and mem_id
+			size = pcer_interface->tensor_unfold(executor, op_idx, &task->op_desc[op_idx],
+				   &task->surface_desc[op_idx], tensor_set, j);  //get dsp flat1 len and mem_id
 			if (size < 0) {
 				dla_error("err:get dsp flat1 len failed!\n");
 				return -1;
 			}
 
 			dsp_tensor->flat1_size = size;
-			dsp_tensor->flat1_addr_offset =
-				flat1_total_len[op_type - DLA_KMD_OP_DSP_0];
-			flat1_total_len[op_type - DLA_KMD_OP_DSP_0] +=
-				dsp_tensor->flat1_size;
+			dsp_tensor->flat1_addr_offset = flat1_total_len[op_type - DLA_KMD_OP_DSP_0];
+			flat1_total_len[op_type - DLA_KMD_OP_DSP_0] += dsp_tensor->flat1_size;
 
 			dsp_tensor_idx[op_type - DLA_KMD_OP_DSP_0]++;
 			dsp_op_cnt++;
@@ -608,31 +540,19 @@ int resolve_dsp_data(struct win_executor *executor)
 		if (executor->dsp_iobuf_cnt[i] == 0) {
 			continue;
 		}
-		executor->dsp_iobuf_virt[i] =
-			kzalloc(executor->dsp_iobuf_cnt[i] *
-					(sizeof(u64) + sizeof(u32)),
-				GFP_KERNEL);
+		executor->dsp_iobuf_virt[i] = kzalloc(executor->dsp_iobuf_cnt[i] *
+					(sizeof(u64) + sizeof(u32)), GFP_KERNEL);
 		if (executor->dsp_iobuf_virt[i] == NULL) {
-			dla_error(
-				"%s, %d, alloc dsp iobuf vir for device %d error.\n",
-				__func__, __LINE__, i);
+			dla_error("%s, %d, alloc dsp iobuf vir for device %d error.\n", __func__, __LINE__, i);
 			goto alloc_err;
 		}
 		size = 0;
-		executor->dsp_iobuf_offset[i] =
-			executor->dsp_iobuf_virt[i] +
-			executor->dsp_iobuf_cnt[i] * sizeof(u64);
+		executor->dsp_iobuf_offset[i] = executor->dsp_iobuf_virt[i] + executor->dsp_iobuf_cnt[i] * sizeof(u64);
 		for (j = 0; j < DSP_KERNEL_MAX_INOUT_TENSOR_NUM; j++) {
 			if (executor->dsp_all_inout[i][j] != 0) {
-				executor->dsp_io[i][j].virt =
-					(u64 *)(executor->dsp_iobuf_virt[i] +
-						size * sizeof(u64));
-				executor->dsp_io[i][j].offset =
-					(u32 *)(executor->dsp_iobuf_offset[i] +
-						size * sizeof(u32));
-				dla_debug("%s, %d, virt[0]=0x%px.\n", __func__,
-					  __LINE__,
-					  executor->dsp_io[i][j].virt);
+				executor->dsp_io[i][j].virt = (u64 *)(executor->dsp_iobuf_virt[i] + size * sizeof(u64));
+				executor->dsp_io[i][j].offset = (u32 *)(executor->dsp_iobuf_offset[i] + size * sizeof(u32));
+				dla_debug("%s, %d, virt[0]=0x%px.\n", __func__, __LINE__, executor->dsp_io[i][j].virt);
 				size += executor->dsp_all_inout[i][j];
 			}
 		}
@@ -643,64 +563,48 @@ int resolve_dsp_data(struct win_executor *executor)
 		}
 		dsp_dev = executor->engine->dsp_dev[i];
 		if (dsp_dev == NULL) {
-			dla_error("%s, %d, dsp device%d is NULL, err.\n",
-				  __func__, __LINE__, i);
+			dla_error("%s, %d, dsp device%d is NULL, err.\n", __func__, __LINE__, i);
 			ret = -ENODEV;
 			goto err_dev;
 		}
 
 		executor->dsp_flat1_set_vaddr[i] = dma_alloc_coherent(
-			dsp_dev, flat1_total_len[i],
-			&executor->dsp_flat1_set_dma[i], GFP_KERNEL);
+								dsp_dev, flat1_total_len[i],
+								&executor->dsp_flat1_set_dma[i], GFP_KERNEL);
 		if (executor->dsp_flat1_set_vaddr[i] == NULL) {
-			dla_error(
-				"%s, %d, dma alloc cohernet for dsp dev%d error.\n",
-				__func__, __LINE__, i);
+			dla_error("%s, %d, dma alloc cohernet for dsp dev%d error.\n", __func__, __LINE__, i);
 			goto alloc_err;
 		}
-		dla_debug("%s, %d, i=%d, flat vaddr=0x%px.\n", __func__,
-			  __LINE__, i, executor->dsp_flat1_set_vaddr[i]);
+		dla_debug("%s, %d, i=%d, flat vaddr=0x%px.\n", __func__, __LINE__, i, executor->dsp_flat1_set_vaddr[i]);
 		executor->dsp_flat1_set_size[i] = flat1_total_len[i];
 		/*get the fd of dsp param buffer*/
-		fd = dla_data_get_fd(executor->driver_context,
-				     executor->mem_handles, NULL,
-				     executor->network->op_config_index);
+		fd = dla_data_get_fd(executor->driver_context, executor->mem_handles, NULL,
+							 executor->network->op_config_index);
 		if (fd < 0) {
-			dla_error(
-				"err:dla_data_get_fd failed,i=%d op_config_index=%d!\n",
-				i, executor->network->op_config_index);
+			dla_error("err:dla_data_get_fd failed,i=%d op_config_index=%d!\n",
+					  i, executor->network->op_config_index);
 			goto alloc_err;
 		}
-		executor->model_dsp_dmabuf[i] =
-			npu_get_dsp_dmabuf(executor->dsp_file[i], fd);
+		executor->model_dsp_dmabuf[i] = npu_get_dsp_dmabuf(executor->dsp_file[i], fd);
 		if (executor->model_dsp_dmabuf[i] == NULL) {
 			dla_error("err:import dsp data dmabuf error!\n");
 			goto alloc_err;
 		}
 	}
 
-	for (op_type = DLA_KMD_OP_DSP_0; op_type <= DLA_KMD_OP_DSP_3;
-	     op_type++) {
+	for (op_type = DLA_KMD_OP_DSP_0; op_type <= DLA_KMD_OP_DSP_3; op_type++) {
 		if (dsp_tensor_idx[op_type - DLA_KMD_OP_DSP_0] == 0) {
 			continue;
 		}
 		tensor_set = executor->tensor_set[op_type];
-		dma_set =
-			executor->dsp_flat1_set_dma[op_type - DLA_KMD_OP_DSP_0];
-		flat_vaddr_set =
-			executor->dsp_flat1_set_vaddr[op_type -
-						      DLA_KMD_OP_DSP_0];
-		data_dma_set =
-			executor->model_dsp_dmabuf[op_type - DLA_KMD_OP_DSP_0]
-				->dma_addr;
-
-		for (i = 0; i < dsp_tensor_idx[op_type - DLA_KMD_OP_DSP_0];
-		     i++) {
+		dma_set = executor->dsp_flat1_set_dma[op_type - DLA_KMD_OP_DSP_0];
+		flat_vaddr_set = executor->dsp_flat1_set_vaddr[op_type - DLA_KMD_OP_DSP_0];
+		data_dma_set = executor->model_dsp_dmabuf[op_type - DLA_KMD_OP_DSP_0]->dma_addr;
+
+		for (i = 0; i < dsp_tensor_idx[op_type - DLA_KMD_OP_DSP_0]; i++) {
 			dsp_tensor = &tensor_set[i];
-			dsp_tensor->flat1_dma_addr =
-				dma_set + dsp_tensor->flat1_addr_offset;
-			dsp_tensor->flat1_vaddr =
-				flat_vaddr_set + dsp_tensor->flat1_addr_offset;
+			dsp_tensor->flat1_dma_addr = dma_set + dsp_tensor->flat1_addr_offset;
+			dsp_tensor->flat1_vaddr = flat_vaddr_set + dsp_tensor->flat1_addr_offset;
 			dsp_tensor->data_dma_addr = data_dma_set;
 		}
 	}
@@ -716,16 +620,13 @@ int resolve_dsp_data(struct win_executor *executor)
 			continue;
 		}
 		if (executor->dsp_flat1_set_vaddr[i] != NULL) {
-			dma_free_coherent(dsp_dev,
-					  executor->dsp_flat1_set_size[i],
-					  executor->dsp_flat1_set_vaddr[i],
-					  executor->dsp_flat1_set_dma[i]);
+			dma_free_coherent(dsp_dev, executor->dsp_flat1_set_size[i],
+					  executor->dsp_flat1_set_vaddr[i], executor->dsp_flat1_set_dma[i]);
 			executor->dsp_flat1_set_vaddr[i] = NULL;
 			executor->dsp_flat1_set_dma[i] = 0;
 		}
 		if (executor->model_dsp_dmabuf[i] != NULL) {
-			npu_put_dsp_dmabuf(executor->dsp_file[i],
-					   &executor->model_dsp_dmabuf[i]);
+			npu_put_dsp_dmabuf(executor->dsp_file[i], &executor->model_dsp_dmabuf[i]);
 			executor->model_dsp_dmabuf[i] = NULL;
 		}
 		if (executor->dsp_iobuf_virt[i] != NULL) {
@@ -755,28 +656,19 @@ int npu_set_dsp_iobuf(struct win_executor *executor, struct host_frame_desc *f)
 			}
 			fd = address[j].devBuf.memFd;
 
-			f->dsp_io_dmabuf[i][j] =
-				npu_get_dsp_dmabuf(executor->dsp_file[i], fd);
+			f->dsp_io_dmabuf[i][j] = npu_get_dsp_dmabuf(executor->dsp_file[i], fd);
 			if (f->dsp_io_dmabuf[i][j] == NULL) {
-				dla_error(
-					"%s, %d, npu get dsp%d dmabuf-%d error.\n ",
-					__func__, __LINE__, i, j);
+				dla_error("%s, %d, npu get dsp%d dmabuf-%d error.\n ", __func__, __LINE__, i, j);
 				return -EINVAL;
 			}
-			dla_debug("%s, %d dspio=%d.\n", __func__, __LINE__,
-				  executor->dsp_io[i][j].io_cnt);
+			dla_debug("%s, %d dspio=%d.\n", __func__, __LINE__, executor->dsp_io[i][j].io_cnt);
 			for (k = 0; k < executor->dsp_io[i][j].io_cnt; k++) {
-				dla_debug("%s, %d, virt=0x%llx.\n", __func__,
-					  __LINE__,
-					  executor->dsp_io[i][j].virt[k]);
+				dla_debug("%s, %d, virt=0x%llx.\n", __func__, __LINE__, executor->dsp_io[i][j].virt[k]);
 				tmp = (u32 *)executor->dsp_io[i][j].virt[k];
 				offset = executor->dsp_io[i][j].offset[k];
-				dla_debug(
-					"%s, %d, offset=0x%x, dma addr=0x%x.\n",
-					__func__, __LINE__, offset,
-					f->dsp_io_dmabuf[i][j]->dma_addr);
-				*tmp = offset + address[j].devBuf.offset +
-				       f->dsp_io_dmabuf[i][j]->dma_addr;
+				dla_debug("%s, %d, offset=0x%x, dma addr=0x%x.\n", __func__, __LINE__, offset,
+						  f->dsp_io_dmabuf[i][j]->dma_addr);
+				*tmp = offset + address[j].devBuf.offset + f->dsp_io_dmabuf[i][j]->dma_addr;
 
 				dla_debug("tmp content=0x%x.\n", *tmp);
 			}
@@ -805,13 +697,11 @@ static int npu_unload_dsp_op(struct win_executor *executor, int idx,
 	}
 	dsp_tensor = &tensor_set[idx];
 	if (dsp_tensor == NULL || (dsp_tensor->handle == 0)) {
-		dla_error(
-			"%s, %d, idx=%d, op_type=%d.\n dsp tensor null or handle is 0, err.\n",
-			__func__, __LINE__, idx, op_type);
+		dla_error("%s, %d, idx=%d, op_type=%d.\n dsp tensor null or handle is 0, err.\n",
+				  __func__, __LINE__, idx, op_type);
 		return -EINVAL;
 	}
-	ret = unload_operator(executor->engine->dsp_dev[dev_id],
-			      dsp_tensor->handle);
+	ret = unload_operator(executor->engine->dsp_dev[dev_id], dsp_tensor->handle);
 	return ret;
 }
 
@@ -828,8 +718,7 @@ void destroy_frame_dsp_info(struct win_executor *executor,
 			if (f->dsp_io_dmabuf[i][j] == NULL) {
 				continue;
 			}
-			npu_put_dsp_dmabuf(executor->dsp_file[i],
-					   &f->dsp_io_dmabuf[i][j]);
+			npu_put_dsp_dmabuf(executor->dsp_file[i], &f->dsp_io_dmabuf[i][j]);
 			f->dsp_io_dmabuf[i][j] = NULL;
 		}
 	}
@@ -856,16 +745,13 @@ void dsp_resource_destroy(struct win_executor *executor)
 		xa_destroy(&executor->dsp_ddr_xrray[i]);
 
 		if (executor->dsp_flat1_set_vaddr[i] != NULL) {
-			dma_free_coherent(dsp_dev,
-					  executor->dsp_flat1_set_size[i],
-					  executor->dsp_flat1_set_vaddr[i],
-					  executor->dsp_flat1_set_dma[i]);
+			dma_free_coherent(dsp_dev, executor->dsp_flat1_set_size[i],
+					executor->dsp_flat1_set_vaddr[i], executor->dsp_flat1_set_dma[i]);
 			executor->dsp_flat1_set_vaddr[i] = NULL;
 			executor->dsp_flat1_set_dma[i] = 0;
 		}
 		if (executor->model_dsp_dmabuf[i] != NULL) {
-			npu_put_dsp_dmabuf(executor->dsp_file[i],
-					   &executor->model_dsp_dmabuf[i]);
+			npu_put_dsp_dmabuf(executor->dsp_file[i], &executor->model_dsp_dmabuf[i]);
 			executor->model_dsp_dmabuf[i] = NULL;
 		}
 
diff --git a/drivers/soc/eswin/ai_driver/npu/engine.c b/drivers/soc/eswin/ai_driver/npu/engine.c
index fa165ac48a2f..fe6d3c1339c8 100644
--- a/drivers/soc/eswin/ai_driver/npu/engine.c
+++ b/drivers/soc/eswin/ai_driver/npu/engine.c
@@ -74,8 +74,7 @@ int set_current(struct win_engine *engine, struct win_executor *executor,
 	struct win_engine *win_engine = engine;
 
 	if (unlikely(win_engine->cur[tiktok] != NULL || executor == NULL)) {
-		dla_error(
-			"BUG: trying to use busy engine or set engine NULL\n");
+		dla_error("BUG: trying to use busy engine or set engine NULL\n");
 		return -1;
 	}
 	win_engine->cur[tiktok] = executor;
@@ -214,16 +213,14 @@ int io_tensor_record(struct win_executor *executor, addrDesc_t *address,
 	switch (address->flag) {
 	case mem_flag_input:
 		if (address->bindId > executor->input_num) {
-			dla_error("%s %d invalid bind_id %d\n", __func__,
-				  __LINE__, address->bindId);
+			dla_error("%s %d invalid bind_id %d\n", __func__, __LINE__, address->bindId);
 			return -1;
 		}
 		*is_io_tensor = address->bindId;
 		return 1;
 	case mem_flag_output:
 		if (address->bindId > executor->output_num) {
-			dla_error("%s %d invalid bind_id %d\n", __func__,
-				  __LINE__, address->bindId);
+			dla_error("%s %d invalid bind_id %d\n", __func__, __LINE__, address->bindId);
 			return -1;
 		}
 		*is_io_tensor = address->bindId + executor->input_num;
@@ -332,16 +329,14 @@ static void frame_done_handler(struct work_struct *w)
 	while (true) {
 		f = engine->frame_done;
 		ret = __atomic_compare_exchange_n(&engine->frame_done, &f, NULL,
-						  true, __ATOMIC_RELAXED,
-						  __ATOMIC_RELAXED);
+						  true, __ATOMIC_RELAXED, __ATOMIC_RELAXED);
 		if (ret) {
 			break;
 		}
 	}
 
 	if (f == NULL) {
-		dla_error("error:%s %d frame_done is null!\n", __func__,
-			  __LINE__);
+		dla_error("error:%s %d frame_done is null!\n", __func__, __LINE__);
 		return;
 	}
 
@@ -368,8 +363,7 @@ static void npu_task_complete_work(struct work_struct *work)
 
 	while (!list_empty(&tmp_list)) {
 		struct host_frame_desc *f;
-		f = list_first_entry(&tmp_list, struct host_frame_desc,
-				     complete_entry);
+		f = list_first_entry(&tmp_list, struct host_frame_desc, complete_entry);
 		list_del(&f->complete_entry);
 		if (f->dump_dtim) {
 			dump_dtim_to_file(engine, f->tiktok);
@@ -412,13 +406,9 @@ static int npu_get_dsp_dev(struct win_engine *engine)
 
 #if (NPU_DEV_SIM == NPU_REAL_ENV)
 		if (!ret) {
-			ret = dsp_attach_sram_dmabuf(
-				engine->dsp_dev[i],
-				nvdla_dev->spram_bobj->dmabuf);
+			ret = dsp_attach_sram_dmabuf(engine->dsp_dev[i], nvdla_dev->spram_bobj->dmabuf);
 			if (ret) {
-				dla_error(
-					"%s, %d, attach dsp sram failed, ret=%d.\n",
-					__func__, __LINE__, ret);
+				dla_error("%s, %d, attach dsp sram failed, ret=%d.\n", __func__, __LINE__, ret);
 				goto err;
 			}
 		}
@@ -518,8 +508,7 @@ int win_engine_init(struct nvdla_device *nvdla_dev, void **arg_engine)
 		INIT_WORK(&engine->dump_file_work[i].work, dump_data_to_file);
 	}
 #elif (NPU_DEV_SIM == NPU_REAL_ENV)
-	engine->dump_op_work_queue =
-		alloc_ordered_workqueue("dump_op_to_file", 0);
+	engine->dump_op_work_queue = alloc_ordered_workqueue("dump_op_to_file", 0);
 	for (i = 0; i < NUM_OP_TYPE; i++) {
 		INIT_WORK(&engine->dump_op_work[i].work, dump_data_to_file);
 	}
@@ -551,8 +540,7 @@ int win_engine_init(struct nvdla_device *nvdla_dev, void **arg_engine)
 
 	ret = npu_get_dsp_dev(engine);
 	if (ret) {
-		dla_error("%s, %d, get dsp device err, ret=%d.\n", __func__,
-			  __LINE__, ret);
+		dla_error("%s, %d, get dsp device err, ret=%d.\n", __func__, __LINE__, ret);
 		devm_kfree(&nvdla_dev->pdev->dev, engine->is_event_source_done[0]);
 		devm_kfree(&nvdla_dev->pdev->dev, engine);
 		return -ENOMEM;
@@ -678,7 +666,7 @@ static void free_executor_tensor_data(struct win_executor *executor)
 			executor->prog_data_size[i] = 0;
 		}
 		if (executor->tensor_set[i] != NULL) {
-			kfree(executor->tensor_set[i]);
+			vfree(executor->tensor_set[i]);
 			executor->tensor_set[i] = NULL;
 		}
 	}
@@ -689,8 +677,7 @@ static void free_executor_tensor_data(struct win_executor *executor)
 		if (cnt[i] < 1) {                                            \
 			continue;                                            \
 		}                                                            \
-		executor->tensor_set[i] =                                    \
-			kzalloc(cnt[i] * sizeof(op##_tensor_t), GFP_KERNEL); \
+		executor->tensor_set[i] = vzalloc(cnt[i] * sizeof(op##_tensor_t)); \
 		size[i] = cnt[i] * sizeof(op##_dev_t);                       \
 		if (executor->tensor_set[i] == NULL) {                       \
 			size[i] = 0;                                         \
@@ -721,26 +708,21 @@ static int pre_setup_op_tensor(void *pexecutor, struct dla_task *task,
 		ret = -EINVAL;
 		return ret;
 	}
-	executor->io_mem_handle_size =
-		(executor->input_num + executor->output_num) *
-		sizeof(addrDesc_t);
-	executor->frame_size =
-		sizeof(struct host_frame_desc) + executor->io_mem_handle_size +
-		(executor->input_num + executor->output_num) * sizeof(u64);
+	executor->io_mem_handle_size = (executor->input_num + executor->output_num) * sizeof(addrDesc_t);
+	executor->frame_size = sizeof(struct host_frame_desc) + executor->io_mem_handle_size +
+				(executor->input_num + executor->output_num) * sizeof(u64);
 	memset(cnt, 0, sizeof(cnt));
 	memset(size, 0, sizeof(size));
 
 	for (i = 0; i < op_num; i++) {
-		dla_detail("i=%d, op_type=%d.\n", i,
-			   task->common_desc[i].op_type);
+		dla_detail("i=%d, op_type=%d.\n", i, task->common_desc[i].op_type);
 		pcer = processor_dla_convert[task->common_desc[i].op_type];
 		//the same processor have how many op.
 		cnt[pcer]++;
 	}
 
 	for (i = IDX_START; i < NUM_OP_TYPE; i++) {
-		dla_info("%s, %d, i=%d, cnt=%d.\n", __func__, __LINE__, i,
-			 cnt[i]);
+		dla_info("%s, %d, i=%d, cnt=%d.\n", __func__, __LINE__, i, cnt[i]);
 	}
 
 	for (i = IDX_START; i < NUM_OP_TYPE; i++) {
@@ -789,15 +771,12 @@ static int pre_setup_op_tensor(void *pexecutor, struct dla_task *task,
 		}
 		if (i == IDX_CONV) {
 			size[i] = cnt[i] * MAX_CONV_PROG_DATA_SIZE;
-			dla_debug("single MAX_CONV_PROG_SIZE=%u.\n",
-				  MAX_CONV_PROG_SIZE);
+			dla_debug("single MAX_CONV_PROG_SIZE=%u.\n", MAX_CONV_PROG_SIZE);
 		}
 
 		executor->prog_data_size[i] = size[i];
 		executor->prog_data_buf_bobj[i] = npu_alloc_dma_addr(
-			executor, size[i], &executor->dma_addr[i], i,
-			GFP_KERNEL);
-
+			executor, size[i], &executor->dma_addr[i], i, GFP_KERNEL);
 		if (executor->prog_data_buf_bobj[i] == NULL) {
 			dla_error("%s, %d, i=%d.\n", __func__, __LINE__, i);
 			ret = -ENOMEM;
@@ -818,8 +797,7 @@ static int extract_input_output_address(struct win_executor *executor)
 	int i, input_num = 0, output_num = 0;
 	addrDesc_t *address = executor->mem_handles->addrlist->addrDesc;
 
-	dla_debug("num_addresses=%d\n",
-		  executor->mem_handles->addrlist->numAddress);
+	dla_debug("num_addresses=%d\n", executor->mem_handles->addrlist->numAddress);
 	for (i = 0; i < executor->mem_handles->addrlist->numAddress; i++) {
 		if (address[i].flag == mem_flag_input) {
 			dla_debug("i=%d,input\n", i);
@@ -831,8 +809,7 @@ static int extract_input_output_address(struct win_executor *executor)
 		}
 	}
 
-	return pre_setup_op_tensor(executor, executor->task, input_num,
-				   output_num,
+	return pre_setup_op_tensor(executor, executor->task, input_num, output_num,
 				   executor->network->num_operations);
 }
 
@@ -849,19 +826,16 @@ int create_executor(struct dla_task *task, struct dla_network_desc *network,
 		return -ENOMEM;
 	}
 
-	dla_debug("executor=0x%px executor_size=%ld\n", executor,
-		  sizeof(struct win_executor));
+	dla_debug("executor=0x%px executor_size=%ld\n", executor, sizeof(struct win_executor));
 
-	executor->dependency_count =
-		kzalloc(network->num_operations, GFP_KERNEL);
+	executor->dependency_count = vzalloc(network->num_operations);
 	if (executor->dependency_count == NULL) {
 		dla_error("%s %d no mem\n", __func__, __LINE__);
 		ret = -ENOMEM;
 		goto err_free0;
 	}
 
-	executor->dump_info.op_idx_list =
-		kzalloc(MAX_OP_NUM * sizeof(u16), GFP_KERNEL);
+	executor->dump_info.op_idx_list = vzalloc(MAX_OP_NUM * sizeof(u16));
 	if (executor->dump_info.op_idx_list == NULL) {
 		dla_error("%s %d no mem\n", __func__, __LINE__);
 		ret = -ENOMEM;
@@ -870,10 +844,9 @@ int create_executor(struct dla_task *task, struct dla_network_desc *network,
 
 	executor->task = task;
 	for (i = 0; i < network->num_operations; i++) {
-		executor->dependency_count[i] =
-			task->common_desc[i].dependency_count;
+		executor->dependency_count[i] = task->common_desc[i].dependency_count;
 		dla_debug("%s,%d, i=%d, cnt=%d.\n", __func__, __LINE__, i,
-			  executor->dependency_count[i]);
+				  executor->dependency_count[i]);
 	}
 
 	executor->network = network;
@@ -918,9 +891,9 @@ int create_executor(struct dla_task *task, struct dla_network_desc *network,
 err_free3:
 	free_executor_tensor_data(executor);
 err_free2:
-	kfree(executor->dump_info.op_idx_list);
+	vfree(executor->dump_info.op_idx_list);
 err_free1:
-	kfree(executor->dependency_count);
+	vfree(executor->dependency_count);
 err_free0:
 	kfree(executor);
 	*m_executor = NULL;
@@ -933,15 +906,15 @@ void executor_clearup(void *arg_executor)
 	int i = 0;
 
 	if (executor->dependency_count != NULL) {
-		kfree(executor->dependency_count);
+		vfree(executor->dependency_count);
 		executor->dependency_count = NULL;
 	}
 	if (executor->dump_info.op_idx_list != NULL) {
-		kfree(executor->dump_info.op_idx_list);
+		vfree(executor->dump_info.op_idx_list);
 		executor->dump_info.op_idx_list = NULL;
 	}
 	if (executor->cfg_seq[IDX_START]) {
-		kfree(executor->cfg_seq[IDX_START]);
+		vfree(executor->cfg_seq[IDX_START]);
 		for (i = IDX_START; i < NUM_OP_TYPE; i++) {
 			executor->cfg_seq[i] = NULL;
 		}
diff --git a/drivers/soc/eswin/ai_driver/npu/parse_dep.c b/drivers/soc/eswin/ai_driver/npu/parse_dep.c
index e21c41146346..8604acd5e545 100644
--- a/drivers/soc/eswin/ai_driver/npu/parse_dep.c
+++ b/drivers/soc/eswin/ai_driver/npu/parse_dep.c
@@ -151,11 +151,8 @@ static npu_dep_info_t *npu_get_dep_info(struct win_executor *executor, u8 pcer,
 		get_npu_info(edma, EDMA, k);
 		break;
 	case IDX_CONV:
-		npu_info = (npu_dep_info_t
-				    *)((char *)executor
-					       ->prog_data_buf_bobj[IDX_CONV] +
-				       MAX_CONV_PROG_DATA_SIZE * k +
-				       sizeof(rdma_dev_com_inf_t));
+		npu_info = (npu_dep_info_t *)((char *)executor->prog_data_buf_bobj[IDX_CONV] +
+				       MAX_CONV_PROG_DATA_SIZE * k + sizeof(rdma_dev_com_inf_t));
 		break;
 	case IDX_SDP:
 		get_npu_info(sdp, SDP, k);
@@ -218,14 +215,11 @@ static void dependency_consumer2producer(struct win_executor *executor,
 
 	for (i = 0; i < op_num; i++) {
 		pcer = processor_dla_convert[task->common_desc[i].op_type];
-		dla_detail("i:%d op_type:%d\n", i,
-			   task->common_desc[i].op_type);
+		dla_detail("i:%d op_type:%d\n", i, task->common_desc[i].op_type);
 		k = pcer_cnt[pcer];
 		npu_info = npu_get_dep_info(executor, pcer, k);
 		if (npu_info == NULL) {
-			dla_error(
-				"get processor %d, op_idx=%d, npu_info is null.\n",
-				pcer, k);
+			dla_error("get processor %d, op_idx=%d, npu_info is null.\n", pcer, k);
 			continue;
 		}
 		if (pcer != IDX_CONV)
@@ -242,29 +236,24 @@ static void dependency_consumer2producer(struct win_executor *executor,
 			           dependency_count[i], npu_info->peer_type);
 		}
 
-		for (j = IDX_SDP; j <= IDX_PDP; j++) {
-			consumer = task->common_desc[i].fused_parent.index;
-			event = task->common_desc[i].fused_parent.event;
-			if (consumer != invalid_op_index &&
-			    event == DLA_EVENT_OP_ENABLED) {
-				npu_set_enable_consumer(npu_info, consumer);
-			}
+		consumer = task->common_desc[i].fused_parent.index;
+		event = task->common_desc[i].fused_parent.event;
+		if (consumer != invalid_op_index &&	event == DLA_EVENT_OP_ENABLED) {
+			npu_set_enable_consumer(npu_info, consumer);
 		}
+
 		pos = 0;
 		for (j = IDX_START; j < NUM_OP_TYPE; j++) {
 			op_type = processor_idx_convert[j];
-			consumer =
-				task->common_desc[i].consumers[op_type].index;
+			consumer = task->common_desc[i].consumers[op_type].index;
 			event = task->common_desc[i].consumers[op_type].event;
 			if (consumer == 0xffff) {
 				continue;
 			}
 
 			if (event == DLA_EVENT_OP_PROGRAMMED) {
-				dla_detail(
-					"i:%d j:%d consumer:%d dependency_count:%d\n",
-					i, j, consumer,
-					dependency_count[consumer]);
+				dla_detail("i:%d j:%d consumer:%d dependency_count:%d\n", i, j, consumer,
+						   dependency_count[consumer]);
 				dependency_count[consumer]--;
 			}
 			if (pcer == IDX_CONV) {
@@ -273,14 +262,11 @@ static void dependency_consumer2producer(struct win_executor *executor,
 
 			if (event == DLA_EVENT_OP_COMPLETED) {
 				if (unlikely(pos >= MAX_KMD_DEPCNT)) {
-					dla_error(
-						"op index:%d dependency cnt over max\n",
-						i);
+					dla_error("op index:%d dependency cnt over max\n", i);
 					BUG_ON(false);
 					return;
 				}
-				npu_set_completion_consumer(npu_info, op_type,
-							    consumer, pos);
+				npu_set_completion_consumer(npu_info, op_type, consumer, pos);
 				pos++;
 			}
 
@@ -290,8 +276,7 @@ static void dependency_consumer2producer(struct win_executor *executor,
 			 */
 		}
 		dependency_count[i]++;
-		dla_debug("i:%d dependency_count:%d\n", i,
-			   dependency_count[i]);
+		dla_debug("i:%d dependency_count:%d\n", i, dependency_count[i]);
 		npu_info->depcnt = dependency_count[i];
 		pcer_cnt[pcer]++;
 	}
@@ -305,17 +290,14 @@ int generate_small_program(struct win_executor *executor)
 	op_num = executor->network->num_operations;
 	executor->total_op_num = op_num;
 
-	executor->cfg_seq[IDX_START] =
-		kzalloc(op_num * sizeof(u16), GFP_KERNEL);
+	executor->cfg_seq[IDX_START] = vzalloc(op_num * sizeof(u16));
 	if (executor->cfg_seq[IDX_START] == NULL) {
-		ret = -ENOMEM;
-		return ret;
+		return -ENOMEM;
 	}
 	dla_debug("%s, %d, op_num=%d.\n", __func__, __LINE__, op_num);
 	ret = config_sequence_setup(executor, op_num, executor->cfg_seq);
 	if (ret < 0) {
-		dla_error("%s %d config_sequence_setup fail\n", __func__,
-			  __LINE__);
+		dla_error("%s %d config_sequence_setup fail\n", __func__, __LINE__);
 		goto err_free1;
 	}
 	dependency_consumer2producer(executor, executor->task, op_num);
@@ -323,7 +305,7 @@ int generate_small_program(struct win_executor *executor)
 	return 0;
 err_free1:
 	memset(executor->cfg_seq[IDX_START], -1, op_num * sizeof(u16));
-	kfree(executor->cfg_seq[IDX_START]);
+	vfree(executor->cfg_seq[IDX_START]);
 	executor->cfg_seq[IDX_START] = NULL;
 	return ret;
 }
@@ -375,9 +357,7 @@ int generate_event_map(struct win_executor *executor)
 		case IDX_EVENT_SINK:
 			event_idx = executor->task->op_desc[i].event_op.index;
 			if (event_idx >= executor->total_event_sink_num) {
-				dla_error(
-					"Event Sink, event_idx(%d) is wrong in model.\n",
-					event_idx);
+				dla_error("Event Sink, event_idx(%d) is wrong in model.\n", event_idx);
 				ret = -EINVAL;
 				goto err_free;
 			}
@@ -386,9 +366,7 @@ int generate_event_map(struct win_executor *executor)
 		case IDX_EVENT_SOURCE:
 			event_idx = executor->task->op_desc[i].event_op.index;
 			if (event_idx >= executor->total_event_source_num) {
-				dla_error(
-					"Event Source, event_idx(%d) is wrong in model.\n",
-					event_idx);
+				dla_error("Event Source, event_idx(%d) is wrong in model.\n", event_idx);
 				ret = -EINVAL;
 				goto err_free;
 			}
@@ -427,9 +405,7 @@ int set_pause_op_done(struct win_executor *executor, kmd_dump_info_t *dump_info)
 		k = pcer_cnt[pcer];
 		npu_info = npu_get_dep_info(executor, pcer, k);
 		if (npu_info == NULL) {
-			dla_error(
-				"get processor %d, op_idx=%d, npu_info is null.\n",
-				pcer, k);
+			dla_error("get processor %d, op_idx=%d, npu_info is null.\n", pcer, k);
 			continue;
 		}
 
@@ -445,8 +421,7 @@ int set_pause_op_done(struct win_executor *executor, kmd_dump_info_t *dump_info)
 			if (op_index == task->common_desc[i].index) {
 				npu_info->notify_op_done = 1;
 				npu_info->pause_op_done = 1;
-				dla_debug("%s, %d, dump op_index:%d\n",
-					  __func__, __LINE__, op_index);
+				dla_debug("%s, %d, dump op_index:%d\n", __func__, __LINE__, op_index);
 				break;
 			}
 		}
@@ -476,9 +451,7 @@ int reset_pause_op_done(struct win_executor *executor)
 		k = pcer_cnt[pcer];
 		npu_info = npu_get_dep_info(executor, pcer, k);
 		if (npu_info == NULL) {
-			dla_error(
-				"get processor %d, op_idx=%d, npu_info is null.\n",
-				pcer, k);
+			dla_error("get processor %d, op_idx=%d, npu_info is null.\n", pcer, k);
 			continue;
 		}
 
-- 
2.54.0

